File: backend/src/main/java/isis/projet/backend/entity/Capteur.java
package isis.projet.backend.entity;

import java.util.ArrayList;
import java.util.List;
import java.util.Optional;

import jakarta.persistence.*;


import lombok.*;

// Un exemple d'entité
// On utilise Lombok pour auto-générer getter / setter / toString...
// cf. https://examples.javacodegeeks.com/spring-boot-with-lombok/
@Getter @Setter @NoArgsConstructor @RequiredArgsConstructor @ToString
@Entity // Une entité JPA
public class Capteur {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(unique=true, name = "grandeur")
    @NonNull
    private String grandeur;

    //obtenir une liste de mesures pour chaque grandeur
    @OneToMany(mappedBy = "capteur")
    private List<Mesure> mesures = new ArrayList<>();

    public Optional<List<Mesure>> getMesures(){
        return Optional.ofNullable(mesures);
    }
}

File: frontend/src/Capteur.js
export default class Capteur {
    constructor(id, grandeur, listeMesures) {
      this._id = id;
      this._grandeur = grandeur;
      this._listeMesures = listeMesures;
    }
    get grandeur() {
      return this._grandeur;
    }
    get id() {
      return this._id;
    }
    get listeMesures() {
      return this._listeMesures;
    }
  }
  